extensions/wikia/ThemeDesigner/ThemeSettings.class.php

```diff
@@ -124,85 +130,83 @@ public function getHistory() {
+			$file = new LocalFile( Title::newFromText( $title, NS_FILE ), RepoGroup::singleton()->getLocalRepo() );
 			$file->upload( $temp_file->getPath(), '', '' );
 			$temp_file->delete( '' );
-			Wikia::invalidateFavicon();
 
-			$settings['favicon-image-url'] = $file->getURL();
-			$settings['favicon-image-name'] = $file->getName();
+			$settings["${$name}"] = $file->getURL();
+			$settings["${$name}-name"] = $file->getName();
 
-			$file->repo->forceMaster();
-			$history = $file->getHistory( 1 );
-			if ( count( $history ) == 1 ) {
-				$oldFaviconFile = [ 'url' => $history[0]->getURL(), 'name' => $history[0]->getArchiveName() ];
+			if ($setSize) {
+				$settings["${$name}-width"] = $file->getWidth();
+				$settings["${$name}-height"] = $file->getHeight();
 			}
-		}
-
-		if ( isset( $settings['wordmark-image-name'] ) && strpos( $settings['wordmark-image-name'], 'Temp_file_' ) === 0 ) {
-			$temp_file = new LocalFile( Title::newFromText( $settings['wordmark-image-name'], 6 ), RepoGroup::singleton()->getLocalRepo() );
-			$file = new LocalFile( Title::newFromText( self::WordmarkImageName, 6 ), RepoGroup::singleton()->getLocalRepo() );
-			$file->upload( $temp_file->getPath(), '', '' );
-			$temp_file->delete( '' );
 
-			$settings['wordmark-image-url'] = $file->getURL();
-			$settings['wordmark-image-name'] = $file->getName();
+			if ($setUserImage) {
+				$imageServing = new ImageServing( null, 120, [ "w" => "120", "h" => "65" ] );
+				$settings["user-${$name}"] = $file->getURL();
+				$settings["user-${$name}-thumb"] = wfReplaceImageServer(
+					$file->getThumbUrl(
+						$imageServing->getCut(
+							$file->getWidth(),
+							$file->getHeight(),
+							"origin"
+						) . "-" . $file->getName()
+					)
+				);
+			}
 
 			$file->repo->forceMaster();
 			$history = $file->getHistory( 1 );
+
 			if ( count( $history ) == 1 ) {
-				$oldFile = [ 'url' => $history[0]->getURL(), 'name' => $history[0]->getArchiveName() ];
+				return [ 'url' => $history[0]->getURL(), 'name' => $history[0]->getArchiveName() ];
 			}
 		}
 
-		if ( isset( $settings['background-image-name'] ) && strpos( $settings['background-image-name'], 'Temp_file_' ) === 0 ) {
-			$temp_file = new LocalFile( Title::newFromText( $settings['background-image-name'], 6 ), RepoGroup::singleton()->getLocalRepo() );
-			$file = new LocalFile( Title::newFromText( self::BackgroundImageName, 6 ), RepoGroup::singleton()->getLocalRepo() );
-			$file->upload( $temp_file->getPath(), '', '' );
-			$temp_file->delete( '' );
+		return null;
+	}
 
-			$settings['background-image'] = $file->getURL();
-			$settings['background-image-name'] = $file->getName();
-			$settings['background-image-width'] = $file->getWidth();
-			$settings['background-image-height'] = $file->getHeight();
+	public function saveSettings( $settings, $cityId = null ) {
+		global $wgCityId, $wgUser;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Refactor this code to rely on RequestContext instead of context-dependent globals. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)